yarn-project/acir-simulator/src/avm/opcodes/control_flow.ts

```diff
@@ -12,7 +15,11 @@ export class Return extends Instruction {
   }
 
   execute(machineState: AvmMachineState, _stateManager: AvmStateManager): void {
-    const returnData = machineState.readMemoryChunk(this.returnOffset, this.returnOffset + this.copySize);
+    // TODO: reconsider this casting
+    const returnData = machineState.memory
```

Comment:
re todo I dont think this is irrational, is it regarding the toBigInt or to Fr

Reply:
Yeah, my point (not explained there) was if we should go through bigint, or if Field should support (and know of) conversion to Fr in the class. But I think I like it better like this. Will remove the comment.